hypha/apply/funds/utils.py

```diff
@@ -38,3 +42,33 @@ def get_default_screening_statues():
             default_no.default = True
             default_no.save()
     return [default_yes, default_no]
+
+
+def get_category_options():
+    """
+    Get all category options to show as filter options in the submission table.
+
+    - Only show options for the Category which has filter_on_dashboard set as True.
+    - And only show options which are used in submissions:
+      - To get this we need to first get all ApplicationSubmission form_data and form_fields.
+      - Then in form_fields we need to check for Category Fields by checking the instance of CategoryQuestionBlock
+      - With the field ids get the correct options selected in form_data
+      - Use the list of correct option to filter options
+
+    return: list of set of suitable category options
+    """
+    ApplicationSubmission = apps.get_model('funds', 'ApplicationSubmission')
+    submission_data = ApplicationSubmission.objects.values_list('form_data', 'form_fields')
+    used_category_options = []
+    for item in submission_data:
+        for field in item[1]:
+            if isinstance(field.block, CategoryQuestionBlock):
```

Comment:
OTF have currently around 6000 submissions, iterating through all the form data takes 9s on my machine. Line 63 3s and line 65 the other 6s.

Reply:
Hm, then this doesn't seem like a good idea to get category questions filters options. _(and also tests not passing in the ci is another issue with this method)_

@frjo, there are couple of work arounds I can think of, basically the problem is to show limited number of options in the filter.(the correct options which are used by submissions, and without looping through all the submission we cant get those options)

- One is to show all the options for category which has filter_on_dashboard `True`, and in order to minimise that more we can also have a filter_on_dashboard check for each options. So along with checking filter_on_dashboard for a category we can also check that for its options and then list those as filter options)

- Second is to have a field used_by_submission to option and when the submission is getting saved for the first time, mark it automatically, but the problem is this won't work for existing submissions. (Note that if we go with this way I would prefer using a field `category_id` instead of `used_by_submission` and checking if its null or not, as category id is used in form_data to save correct options so it will be helpful.)

Reply:
I think we can simply show all options where the category has filter_on_dashboard set too true. No need to have that checkbox on each option.

If staff happens to pick an option with no submission they get an empty table, no biggy.